.github/workflows/ci.yml

```diff
@@ -0,0 +1,75 @@
+name: Build macOS, Ubuntu, Windows
+
+on:
+  push:
+    tags:
+      - '[0-9]+.[0-9]+.[0-9]+*'
+      - 'v[0-9]+.[0-9]+.[0-9]+*'
+
+  workflow_dispatch:
+
+jobs:
+  release:
+    name: Release ${{ matrix.target }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-latest
+            profile: release+static
+            target: x86_64-unknown-linux-musl
+            needs_musl: true
+
+          - os: ubuntu-latest
+            profile: release+static+aarch64
+            target: aarch64-unknown-linux-musl
+            needs_musl: true
+
+          - os: macos-11
+            profile: release+mac
+            target: x86_64-apple-darwin
+
+          - os: macos-11
+            profile: release+mac+aarch64
+            target: aarch64-apple-darwin
+
```

Comment:
We can reduce this matrix to have just windows and mac since Jenkins builds Linux

Reply:
I get that that is more efficient in terms of resources used- to me it seems desirable to build everything in the same place, maybe even remove the build from jenkins if we are already doing it here- though I have not looked into the implications of that.

Reply:
Up to you, Linux is the least expensive platform so it's fine to leave it.  However, I don't think we are ready to move off of Jenkins as I don't think our unit/integration tests will run... but it's worth trying at least once.

Reply:
Help me understand the problem: is it the case that there is a machine that runs Jenkins only and that tests are fed to that machine? I seem to recall that being part of our lab walkthrough. <- I am under the impression that if this is true, tests do not presently have a way of running on GHA.

If the above is true, can I create a ticket to look into how to setup a machine dedicated to performing GHA tests?
If the above is false, why would GHA not be capable of running tests?

Reply:
> Help me understand the problem: is it the case that there is a machine that runs Jenkins only and that tests are fed to that machine? I seem to recall that being part of our lab walkthrough. <- I am under the impression that if this is true, tests do not presently have a way of running on GHA.

There are a number of different machines (which are called "executors" or "nodes" in Jenkins) connected to Jenkins, some of them are allocated dynamically (via AWS EC2) and some of them are persistent.  The ones you saw in the lab are persistent, most of the time though we are using the dynamically allocated executors/nodes.  These are preconfigured with credentials that allow them to access our cloud infrastructure (the `esthri-test` bucket in particular).

This mix of dynamic/persistent machines for running tests and compilations is similar on GitHub actions too, except on GitHub Actions the equivalent concept is called a "runner".

> If the above is true, can I create a ticket to look into how to setup a machine dedicated to performing GHA tests?

We have an AWS IAM credential that allows us to access some of our cloud resources from GitHub Actions.  I believe the work would be to adjust this credential so that it can access the `esthri-test`.

Reply:
think this might be helpful
https://github.com/swift-nav/ICBINS/blob/14d8a704e7e42a3d12b2f711b5455df99d9c67e9/.github/workflows/publish.yml#L89-L94